### PR TITLE
Generic autonomous command setter

### PR DESCRIPTION
**Pull request by aschokking**
* Removes boiler plate code with setting auto commands
* Ensures the commands are only constructed at initialize time of the command so any properties being read in the command group constructors are applied each time the button is pressed
* Refactor alliance color logic to the command group constructor

### CODE REVIEW

File: Competition2017/src/competition/subsystems/autonomous/selection/GenericAutonomousCommandSetter.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            Provider<T> commandProvider
+            ) {
+        super(autonomousCommandSelector);
+        this.commandProvider = commandProvider;
+    }
+
+    @Override
+    public void initialize() {
+        this.autonomousCommandSelector.setCurrentAutonomousCommand(this.commandProvider.get());
+
```
Comment (JohnGilb) [inline Competition2017/src/competition/subsystems/autonomous/selection/GenericAutonomousCommandSetter.java]: needs a `log.info("Initializing")` or something along those lines

Comment (aschokking) [reply Competition2017/src/competition/subsystems/autonomous/selection/GenericAutonomousCommandSetter.java]: added

File: Competition2017/src/competition/subsystems/autonomous/selection/GenericAutonomousCommandSetter.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    @Override
+    public void initialize() {
+        T autonomousCommand = this.commandProvider.get();
+        log.info("Setting autonomous command to " + autonomousCommand.getName());
+        this.autonomousCommandSelector.setCurrentAutonomousCommand(autonomousCommand);
+        
+    }
+
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/autonomous/selection/GenericAutonomousCommandSetter.java:28]: Remove superfluous whitespace

File: Competition2017/src/competition/subsystems/autonomous/ShootAndDriveAcrossBaseLineCommandGroup.java [hunk 3]
```diff
@@ -46,6 +47,13 @@ public ShootAndDriveAcrossBaseLineCommandGroup(XPropertyManager propManager,
         this.addSequential(shiftCommand, 0.1);
         
         this.setInitialHeading = setHeading;
+        log.info("Setting alliance to " + poseSubsystem.getAllianceColor() + " alliance");
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/autonomous/ShootAndDriveAcrossBaseLineCommandGroup.java:50]: This isn't something the pose subsystem should be responsible for.

Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/autonomous/ShootAndDriveAcrossBaseLineCommandGroup.java:50]: This logging shouldn't be done in individual commands; can we move this?

**Review by WasabiFan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**